### PR TITLE
Upgrade html-to-react dependency and fix type errors.

### PR DESCRIPTION
**Pull request by na9da**
### What this PR does

Upgrades `html-to-react` to latest version and fix the type errors. This is to fix the CI build errors.

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/ReactViews/Custom/ChartCustomComponent.ts [hunk 1]
```diff
@@ -338,7 +338,8 @@ export default abstract class ChartCustomComponent<
    * column contains a `<chart>`?
    * @param node The node to test
    */
-  private isFirstColumnOfChartRow(node: DomElement): boolean {
+  private isFirstColumnOfChartRow(_node: DomElement): boolean {
+    const node = _node as any;
```
Comment (na9da) [inline lib/ReactViews/Custom/ChartCustomComponent.ts]: Casting to `any` here and below as I think it is not worth the effort typing this properly. The nested typing is gnarly!

File: package.json [hunk 2]
```diff
@@ -85,14 +84,13 @@
     "d3-transition": "^1.0.0",
     "d3-zoom": "^1.8.3",
     "dateformat": "^3.0.3",
-    "domhandler": "^2.4.2",
```
Comment (tephenavies) [inline package.json:88]: I think we shouldn't import "domhandler" without having a direct dependency on "domhandler".

**Review by tephenavies — CHANGES_REQUESTED** [review]

**Comment by zoran995** [comment]
@na9da I am afraid that this could cause the same issue as #5468 although I might be missing something, check the https://github.com/TerriaJS/terriajs/pull/5468#issuecomment-832607495

**Comment by na9da** [comment]
Thanks @zoran995 for pointing that out. Considering this and also @steve9164's comment, I think we should define the type locally and use that instead of depending on domhandler. We are currently importing it only for the type definition. Upgrading `html-to-react` itself is not a problem though.

**Comment by na9da** [comment]
Sorry that, I overwrote the previous changes.

**Review by tephenavies — APPROVED** [review]

**Pull request merged**